src/Adapter/Attribute/Repository/AttributeRepository.php

```diff
@@ -82,4 +84,128 @@ public function assertAllAttributesExist(array $attributeIds): void
             throw new AttributeNotFoundException('Some of provided attributes does not exist');
         }
     }
+
+    /**
+     * @param ProductId $productId
+     * @param LanguageId $langId
+     *
+     * @return array[]
+     */
+    public function getAttributesInfoByProductId(ProductId $productId, LanguageId $langId): array
+    {
+        $associatedProductAttributes = $this->getProductAttributes($productId->getValue());
+        $attributeIds = array_unique(array_map(function (array $attributeByCombination): int {
+            return (int) $attributeByCombination['id_attribute'];
+        }, $associatedProductAttributes));
+
+        return $this->getAttributesInformation($attributeIds, $langId->getValue());
+    }
+
+    /**
+     * @param int[] $combinationIds
+     * @param LanguageId $langId
+     *
+     * @return array<int, array<int, mixed>>
+     */
+    public function getAttributesInfoByCombinationIds(array $combinationIds, LanguageId $langId): array
+    {
+        $attributeCombinationAssociations = $this->getAttributeCombinationAssociations($combinationIds);
+
+        $attributeIds = array_unique(array_map(function (array $attributeByCombination): int {
```

Comment:
Look similar to above

Reply:
Yes in this case it is necessary we need both side of the association

For product however I simplified the query into a single SQL one